### DISCUSSION TITLE
Fail2ban not blocking

### DISCUSSION THREAD

**Discussion by houseofdreams** [fail2ban/fail2ban#3781] [General]
Hi. I installed and configured Fail2ban to block certain "visitors" to my webserver (nginx). It detects the things I want to block and enters them in the UFW firewall, but still they can connect to the server?

This is the jail.local
_[nginx]
enabled = true
port = http,https
filter = nginx
logpath = /var/log/nginx/access.log
banaction = ufw
banaction_allports = ufw
maxretry = 1
findtime = 360
bantime = 36000
ignoreip = 127.0.0.1/8 10.0.0.0/8_

I tested it from my phone using a cellular connection, the IP address is picked up by fail2ban and listed in ufw as REJECT IN, so I assume that IP should not be able to connect to the webserver anymore?

Am I forgetting something? Is there some kind of action to take to "kill" active states or something like that?

**Comment by sebres** [comment]
We have a wiki for that, may be you'd find the answer there (take a look at answer to 2nd Q) - https://github.com/fail2ban/fail2ban/wiki/How-fail2ban-works .

> Is there some kind of action to take to "kill" active states or something like that?

Well conditionally it may be also true - if firewall (net-filter rules, whatever) only checks incoming connections and requests coming over keep-alive connection, the packets of further requests pass the rules of firewall and have been not blocked (simply because the connection is already established and not new incoming connection)... at least unless nginx (or client) doesn't close the connection.
But it depends on firewall configuration, the backend (and native net-filter) used by firewall internally, the behavior of web-server (where it allow to use keep-alive connections and by which errors or constellations they get closed), etc.

In case of ufw, take a look at option `kill-mode` (off by default) :
https://github.com/fail2ban/fail2ban/blob/59c5e78ce90a49a6f5fb325d35199923c9c65f7c/config/action.d/ufw.conf#L35-L43

Also note there may be alternative conditions, like the packets for further request already arrived through net-filter and queued on the web-server side (before backend or authentication processing), so would simply take place as long as web-server answered previous request.
